src/server/graphql/models/Project/projectMutation.js

```diff
@@ -52,25 +51,40 @@ export default {
         const [userId] = teamMemberId.split('::');
         newProject.userId = userId;
       }
-      await r.table('Project').get(id).update(newProject)
-        .do(() => {
-          return r.table('ProjectHistory')
-            .between([id, r.minval], [id, r.maxval], {index: 'projectIdUpdatedAt'})
-            .orderBy({index: 'projectIdUpdatedAt'})
-            .nth(-1)
-            .default({updatedAt: 0})
-            .do((lastDoc) => {
-              return r.branch(
-                lastDoc('updatedAt').gt(r.epochTime((now - DEBOUNCE_TIME) / 1000)),
-                r.table('ProjectHistory').get(lastDoc('id')).update(mergeDoc),
-                r.table('ProjectHistory').insert(lastDoc.merge(mergeDoc, {id: shortid.generate()}))
-              );
-            });
-        });
+      const projectUpdatePromise = r.table('Project').get(projectId).update(newProject);
+      const dbWork = [projectUpdatePromise];
+      // if this is just a sort update, don't bother writing to the history
+      if (Object.keys(updatedProject).length === 2 && (teamSort !== undefined || userSort !== undefined)) {
+        const projectHistoryPromise = r.table('ProjectHistory')
+          .between([projectId, r.minval], [projectId, r.maxval], {index: 'projectIdUpdatedAt'})
+          .orderBy({index: 'projectIdUpdatedAt'})
+          .nth(-1)
+          .default({updatedAt: r.epochTime(0)})
+          .do((lastDoc) => {
+            return r.branch(
+              lastDoc('updatedAt').gt(r.epochTime((now - DEBOUNCE_TIME) / 1000)),
+              r.table('ProjectHistory').get(lastDoc('id')).update(mergeDoc),
+              r.table('ProjectHistory').insert(lastDoc.merge(mergeDoc, {id: shortid.generate()}))
+            );
+          });
+        dbWork.push(projectHistoryPromise);
+      }
       if (rebalance) {
-        // we could possibly combine this into the rebalance if we did a resort on the server, but separate logic is nice
-        await rebalanceProject(rebalance, teamId);
+        const rebalanceField = teamSort !== undefined ? 'teamSort' : 'userSort';
+        const rebalanceCountPromise = await r.table('Project')
```

Comment:
fun fact: i could turn this into only 1 hit to the DB by using offsetsOf, but that is O(n*logn) (for each insert n, find the index of that id in an array). I feel like there still might be a tricky way to use `fold`, but i think in this case, doing more work on the servers is better than doing something expensive on our DB.